Codegen overview doc (#657)
avohq/docs: pages/implementation/explicit-null-in-codegen.mdx

One of the ways we do it is with the required values for the `Sometimes Sent` parameters.

> The fact that we require to explicitly pass a `null` value for every `Sometimes Sent` property when it's not sent allows the compiler to catch a very common implementation related data bugs.

Author: ```suggestion
The fact that we require an explicit null value to be passed for every Sometimes Sent property when it's not sent allows the compiler to catch very common implementation related data bugs.
```
